src/libstore/build/outer-derivation-goal.hh

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "parsed-derivations.hh"
+#include "lock.hh"
+#include "store-api.hh"
+#include "pathlocks.hh"
+#include "goal.hh"
+
+namespace nix {
+
+struct DerivationGoal;
+
+struct OuterDerivationGoal : public Goal
```

Comment:
How about

```suggestion
struct CreateDerivationGoal : public Goal
```

Reply:
Not sure. It's a trampoline that kicks off both steps, not just the first one to get the derivation. If we had a general "serial composition" combinator, I would want to use that.

Reply:
Ah, right. `CreateDerivationAndRealiseGoal`?

Kinda weird that we have "derivation goal" in the first place if its goal isn't to produce a derivation, but that can be renamed to `RealisationGoal` later.